File: School source codes/Lab 11/Radio.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Radio {
  private int station {get;set;}
  private int volume {get;set;}
  private bool turnton {get;set;}
  public Radio(){
    station = 1;
    volume = 1;
    turnton= false;
  }
  public void turnOn()
    {turnton = true;}
  public void turnOff()
    {turnton = false;}

  public int GetStation()
    {return station;}
  public int GetVolume()
    {return volume;}
  public void StationUp()
    {station++;}
    public void StationUp(int x)
    {station = station + x;}
  public void StationDown()
    {station--;}
  public void StationDown(int x)
    {station = station - x;}
  public void VolumeUp()
    {volume++;}
  public void VolumeUp(int x)
    {volume = volume + x;}
  public void VolumeDown()
    {volume--;}
  public void VolumeDown(int x)
    {volume = volume - x;}

  public override string ToString() {
    if (turnton == false)
      {return "The radio is off.";}
    return "The radio station is " + station + " and the volume level is " + volume + ".";
  }
}

class MainClass {
  public static void Main() {
    Radio test = new Radio();
    test.turnOn();
    Console.WriteLine(test);
    test.VolumeUp(3);
    Console.WriteLine(test);
    test.StationUp(5)
    Console.WriteLine(test);
    test.VolumeDown();
    Console.WriteLine(test);
    test.StationUp(3);
    Console.WriteLine(test);
    test.turnOff();
    Console.WriteLine(test);
    test.VolumeUp(2);
    Console.WriteLine(test);
    test.StationDown(2);
    Console.WriteLine(test);
  }
}

File: School source codes/Lab 11/Radio.rb
class Radio
  def initialize()
    @station,@volume,@turnton = 1,1,false
  end
  def turnOn
    @turnton = true
  end
  def turnOff
    @turnton = false
  end
  def GetStation
    @station
  end
  def GetVolume
    @station
  end
  def StationUp(x = 1)
    @station += x
  end
  def StationDown(x = 1)
    @station -= x
  end
  def VolumeUp(x = 1)
    @volume += x
  end
  def VolumeDown(x = 1)
    @volume -= x
  end
  def to_s
    if (@turnton == false)
      "The radio is off."
    else 
      "The radio station is #{@station} and the volume level is #{@volume}."
    end
  end
end
test = Radio.new
test.turnOn
puts test
test.VolumeUp(3)
puts test
test.StationUp(5)
puts test
test.VolumeDown
puts test
test.StationUp(3)
puts test
test.turnOff
puts test
test.VolumeUp(2)
puts test
test.StationDown(2)
puts test